apps/aechannel/src/aesc_offchain_update.erl

```diff
@@ -208,15 +210,28 @@ apply_on_trees(Update, Trees0, OnChainTrees, OnChainEnv, Round, Reserve) ->
                        abi_version = ABIVersion, amount = Amount,
                        call_data = CallData, call_stack = CallStack,
                        gas_price = GasPrice, gas = Gas} ->
-            Caller = account_pubkey(CallerId),
+            Height = aetx_env:height(OnChainEnv),
             ContractPubKey = contract_pubkey(ContractId),
-            Trees1 = remove_tokens(Caller, Amount, Trees0, Reserve),
+            Trees1 = remove_tokens(account_pubkey(CallerId), Amount, Trees0, Reserve),
             Trees2 = add_tokens(ContractPubKey, Amount, Trees1),
             Call = aect_call:new(CallerId, Round, ContractId, Round, GasPrice),
+            PostLima =
```

Comment:
1. What should happen if a pre-lima deployed contract is called after lima.
2. What should happen if a post lima deployed contract calls an pre lima deployed contract.
3. What should happen if a pre lima deployed contract calls an post lima deployed contract.
4. How many existing channels https://roma-net.mdw.aepps.com/channels are currently using smart contracts ;) ?